DeBooxUpx.py

```diff
@@ -1,301 +1,282 @@
+        'MODEL': 'SP_NoteS',
+        'KEY': '3992B2A532F9ABDCB953EDAD7EC95FFF',
+        'IV': '12C928CF2C665AF83B50361C80B88297'
     },
-    'NovaAir2': {
-        "MODEL": "NovaAir2",
-        "STRING_SETTINGS": "EYuHsXc/WKloIWi3HjlYe61zD/eKz2VLD9iU60eklkqS",
-        "STRING_UPGRADE": "aoPzunZMWqjtRuYTeYxTMpLvLinMCLokDsAWnnE6BlRa",
+    'Tab10CPro': {
+        'MODEL': 'Tab10CPro',
+        'KEY': 'D999393AEAA81119AC0F8C8C1EA11089',
+        'IV': 'CD7894A509490BAF2BAA129686A083E4'
     },
     'Tab13': {
-        "MODEL": "Tab13",
-        "STRING_SETTINGS": "KUYQA1AeW/rKHFqJNQarrSS0N34TOomnnqKlAdoYNtt3",
-        "STRING_UPGRADE": "XkcfAVdtX/83wyPMFgZsmLrQuGRX2z7UYKxt1uTKE5aB",
+        'MODEL': 'Tab13',
+        'KEY': '1072DE10B43097036B98B9021F308FB0',
+        'IV': 'F180C655AABF3EB02D9E45B9FD5379B2'
+    },
+    'Tab8': {
+        'MODEL': 'Tab8',
+        'KEY': 'ECC953580D49BAD70D9DBA526EC091D2',
+        'IV': '46085FF59CCE685B38FB651876E8964B'
+    },
+    'TabUltra': {
+        'MODEL': 'TabUltra',
+        'KEY': 'A01FF8E0F4D139FC75FAA49D61E12E9C',
+        'IV': 'EC94D4BA3550734EA8C69DD726444DE3'
     },
     'TabUltraC': {
-        "MODEL": "TabUltraC",
-        "STRING_SETTINGS": "RZFzfJzyeok0sNXRsM4tyY5MO3C+x57v8vPIMcNdki29",
-        "STRING_UPGRADE": "RuZ3DerxC4zyZynbM4EEX5utqLwVCkxUQgc8S7kaY1B5",
+        'MODEL': 'TabUltraC',
+        'KEY': '2DB550268389D895AACB781AB3515DCB',
+        'IV': '13FDC3C394EAB6323BF42428CD275AA2'
     },
-    'Lomonosov-ru': {
-        "MODEL": "Lomonosov",
-        "STRING_SETTINGS": "d/ta4G+/WIb9o0nlZe4/54Ika8QIc7edI7yEBxC3Yy3d",
-        "STRING_UPGRADE": "A/kr526/LYt2/sk08AdWiFnoCJngYBQnr6GvFbKwbsDW",
-        "STRING_LOCAL": "HK19o0ujB9xYyCA9RvOjIXLSJwuL5iER5xvVN6Pu"
+    'TabX': {
+        'MODEL': 'TabX',
+        'KEY': '28E3BE85609F0D5CAA3796643512AAA2',
+        'IV': '8952832CBE65133CF531C978B7AC0696'
+    },
+    'livingstone-ru': {
+        'MODEL': 'LIVINGSTONE',
+        'KEY': 'CDA476FB48B341F03C3217EBDFCA1BFA',
+        'IV': 'B642BDF244FCC321E1A62BBE0EE4C15A'
     }
 }
 
-
 class DeBooxUpx:
     blockSize: int = 2**12  # 4KiB
 
     def __init__(self,
                  MODEL: str,
-                 STRING_SETTINGS: str,
-                 STRING_UPGRADE: str,
-                 STRING_LOCAL: str = ''):
-        tmpKey: bytes = MD5.new((MODEL * 2).encode()).digest()[:8]
-        self.key: bytes = bytes.fromhex(
-            self.decryptStr(tmpKey, STRING_SETTINGS))
-        self.iv: bytes = bytes.fromhex(self.decryptStr(tmpKey, STRING_UPGRADE))
-        self.path: str = self.decryptStr(tmpKey, STRING_LOCAL)
+                 KEY: str,
+                 IV: str):
+        self.key: bytes = bytes.fromhex(KEY)
+        self.iv: bytes = bytes.fromhex(IV)
 
-    @staticmethod
-    def decryptStr(tmpKey: bytes, string: str) -> str:
-        cipher = DES.new(tmpKey, DES.MODE_CFB, iv=b'\xff' * 8, segment_size=64)
-        return cipher.decrypt(b64decode(string)).decode().strip()
```

Comment:
I think it is useful to keep the api to decrypt the older strings, so that people can use it to decrypt older strings.
For example, a method similar to this which returns a dict `{"MODEL": model, "KEY": key, "IV": iv}`:
```python
    @staticmethod
    def decryptOlderStr(MODEL: str, STRING_SETTINGS: str, STRING_UPGRADE: str) -> Dict[str, str]:
        ...
```

Reply:
or a method which accepts both DES-encrypted and plain ones (not limited to the DES-encrypted strings) and then returns a dict available in `boox_strings`.
If so, users who don't know the details about the length or something else of the strings can get the AES keys easily.